src/sensai/data_transformation.py

```diff
@@ -181,6 +270,9 @@ def __init__(self, columns: Optional[Union[str, Sequence[str]]], categories: Uni
         :param ignoreUnknown: if True and an unknown category is encountered during transform, the resulting one-hot
             encoded columns for this feature will be all zeros. if False, an unknown category will raise an error.
         """
+        super().__init__()
+        self.columns = columns
```

Comment:
I really don't like the addition of new public members for informational purposes only, because
1. It's not immediately clear to even programmers of the class that the members are irrelevant for the functionality of the class (as the public status makes them appear more important than the actual members that do impact the functionality)
2. It's not clear to users of the class what the effect of changing the member is
3. (Minor) `summary` will raise an exception when called on previously persisted instances.

Why don't we keep all such parameters in a dictionary called, for example, `_consParams` or `_paramInfo`.

Reply:
added `_paramInfo` to the base class and removed columns from the public interface